File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/Equipment.java
package com.habitrpg.android.habitica.models.inventory;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Equipment extends RealmObject {

    public double value;
    public String type;
    @PrimaryKey
    public String key;
    public String klass;
    public String specialClass;
    public String index;
    public String text;
    public String notes;
    public int con, str, per;
    @SerializedName("int")
    public int _int;
    public Boolean owned;

    public Equipment() {
    }

    public double getValue() {
        return value;
    }

    public void setValue(double value) {
        this.value = value;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getKlass() {
        return klass;
    }

    public void setKlass(String klass) {
        this.klass = klass;
    }

    public String getSpecialClass() {
        return specialClass;
    }

    public void setSpecialClass(String specialClass) {
        this.specialClass = specialClass;
    }


    public String getIndex() {
        return index;
    }

    public void setIndex(String index) {
        this.index = index;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public int getCon() {
        return con;
    }

    public void setCon(int con) {
        this.con = con;
    }

    public int getStr() {
        return str;
    }

    public void setStr(int str) {
        this.str = str;
    }

    public int getPer() {
        return per;
    }

    public void setPer(int per) {
        this.per = per;
    }

    public int get_int() {
        return _int;
    }

    public void set_int(int _int) {
        this._int = _int;
    }

    public Boolean getOwned() {
        return owned;
    }

    public void setOwned(Boolean owned) {
        this.owned = owned;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/inventory/Equipment.kt
package com.habitrpg.android.habitica.models.inventory

import com.google.gson.annotations.SerializedName
import com.habitrpg.android.habitica.models.BaseMainObject
import io.realm.RealmModel
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Equipment : RealmObject(), BaseMainObject {

    var value: Double = 0.toDouble()
    var type: String? = ""

    @PrimaryKey
    var key: String? = ""
    var klass: String = ""
    var specialClass: String = ""
    var index: String = ""
    var text: String = ""
    var notes: String = ""
    var con: Int = 0
    var str: Int = 0
    var per: Int = 0
    var released: Boolean = true

    @SerializedName("int")
    var _int: Int = 0
    var owned: Boolean? = null
    var twoHanded = false
    var mystery = ""
    var gearSet = ""

    override val realmClass: Class<out RealmModel>
        get() = Equipment::class.java
    override val primaryIdentifier: String?
        get() = key
    override val primaryIdentifierName: String
        get() = "key"
}
